### PR TITLE
task/27085 approve user's request

### PR DESCRIPTION
**Pull request by chinhhq-0603**
https://edu-redmine.sun-asterisk.vn/issues/27085

### CODE REVIEW

File: app/views/requests/_deny_button.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<button type="button" class="btn btn-danger btn-sm" data-toggle="modal" data-toggle="tooltip" data-target="#myModal" data-placement="right" title = <%= t ".deny_request" %>><i class="fas fa-times fa-1x"></i></button>
+<div class="modal fade" id="myModal" role="dialog">
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal">&times;</button>
+        <h4 class="modal-title"> <%= t(".deny_reason") %> </h4>
```
Comment (matTroiCuaHieu) [inline app/views/requests/_deny_button.html.erb]: ( )

File: app/views/requests/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+<%= f.hidden_field :user_id, value: current_user.id %>
+<%= f.hidden_field :reason, value: current_user.id %>
+<%= f.hidden_field :status, value: current_user.id %>
+<%= f.hidden_field :approver, value: current_user.id %>
+<%= f.label :deny_reason, "deny_reason" %>
+<div class="field">
+<%= f.text_area :reason, placeholder: t(".reason"), required: "true" %>
+</div>
+<%= f.submit t(".update"), class: "btn btn-primary" %>
+<% end %>
```
Comment (matTroiCuaHieu) [inline app/views/requests/_form.html.erb]: end line

File: app/views/requests/new.html.erb [hunk 3]
```diff
@@ -37,6 +67,24 @@
             <%= f.submit t(".submit"), class: "btn btn-primary" %>
           <div>
         <% end %>
+      <% else %>
+        <h1><%= t "._create_request" %></h1>
+        <%= form_for @request do |f| %>
+          <%= render "shared/error_messages", object: f.object %>
+          <h3><%= t(".location") %>: <%= params[:location_id] %></h3>
```
Comment (matTroiCuaHieu) [inline app/views/requests/new.html.erb]: ( )

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -34,7 +34,7 @@
         <% if current_user.admin? %>
           <td>
             <% unless user.admin? %>
-              <%= link_to user, method: :delete, data: { confirm: t(".you_sure") } do %>
+              <%= link_to user, method: :delete, data: { confirm: t(".you_sure") }, :remote => "true", :class => 'user_delete' do %>
                 <i class="fas fa-trash-alt fa-1x "data-toggle="tooltip" data-html="true" title=<%= t(".delete") %>></i>
```
Comment (matTroiCuaHieu) [inline app/views/users/index.html.erb]: ( )

File: app/views/users/index.html.erb [hunk 5]
```diff
@@ -34,7 +34,7 @@
         <% if current_user.admin? %>
           <td>
             <% unless user.admin? %>
-              <%= link_to user, method: :delete, data: { confirm: t(".you_sure") } do %>
+              <%= link_to user, method: :delete, data: { confirm: t(".you_sure") }, :remote => "true", :class => 'user_delete' do %>
```
Comment (matTroiCuaHieu) [inline app/views/users/index.html.erb]: single quote

**Comment by hoangquangchinh** [comment]
Screencast-from-28-08-2020-15_10_27

File: app/views/locations/index.html.erb [hunk 6]
```diff
@@ -34,7 +34,7 @@
                 <div><%= location.requests.last.plane_code %></div>
               </div>
               <div class="col-sm-4 table-item-action">
-                <%= link_to new_request_path do %>
+                <%= link_to new_request_path locale, plane_id: location.requests.where(status: "accepted").last.plane_id, status: location.requests.where(status: "accepted").last.plane.status do %>
                   <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%= t(".request") %>></i>
```
Comment (matTroiCuaHieu) [inline app/views/locations/index.html.erb]: ( )

File: app/views/locations/index.html.erb [hunk 7]
```diff
@@ -50,7 +50,7 @@
                 <div><%= t ".status.empty" %></div>
               </div>
               <div class="col-sm-4 table-item-action">
-                <%= link_to new_request_path do %>
+                <%= link_to new_request_path locale, location_id: location.id do %>
                   <i class="far fa-calendar-plus fa-2x" data-toggle="tooltip" data-placement = "top" title= <%= t(".request") %>></i>
```
Comment (matTroiCuaHieu) [inline app/views/locations/index.html.erb]: ( )

**Comment by chinhhq-0603** [comment]
Screencast from 31-08-2020 14_39_01

**Comment by chinhhq-0603** [comment]
Screenshot from 2020-08-31 16-21-50

File: app/views/requests/_deny_button.html.erb [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+      <div class="modal-body">
+        <%= form_tag request_path(request, status: Settings.request.status.denied), method: :patch do |f| %>
+          <%= text_field_tag :deny_reason,"", :class => "textfield-deny-reason" %>
+          <p><%= submit_tag("Submit",:class => "btn btn-success btn-reason-submit" ) %></p>
+        <% end %>
+      </div>
+    </div>
+  </div>
+</div>
+
```
Comment (matTroiCuaHieu) [inline app/views/requests/_deny_button.html.erb]: end

File: app/views/requests/index.html.erb [hunk 9]
```diff
@@ -36,7 +36,7 @@
             <% end %>
           <% else %>
             <%= link_to request do %>
-              <span class="label label-success fa-1x"><%=t ".rejected" %></span>
+              <span class="label label-default fa-1x"><%=t ".rejected" %></span>
             <% end %>
           <% end %>
         <% end %>
```
Comment (matTroiCuaHieu) [inline app/views/requests/index.html.erb:42]: <% t ".rejected" %>

File: app/views/requests/new.html.erb [hunk 10]
```diff
@@ -1,6 +1,36 @@
 <div class="card-request">
   <div class="card-request-body">
-    <% if request.referrer.include? Settings.request.referrer.planes %>
+    <% if request.referrer&.include? Settings.request.referrer.planes %>
+      <% if params[:status].eql? Settings.plane.status.ready %>
+        <h1><%= t(".change") %> <%= params[:plane_id] || @plane_id %> <%= t(".into_gara") %></h1>
```
Comment (matTroiCuaHieu) [inline app/views/requests/new.html.erb]: ( )

**Pull request merged**